examples/update_flow_control/README.md

```diff
@@ -0,0 +1,184 @@
+one thread is responsible for managing the Mender D-Bus API communication, and
+another is responsible for dealing with the user input:
+
+* `do_handle_update_control_map`: initializes the D-Bus communication with the
+  Mender client, sets a "pause all" Update Control Map, and enters the main
+  event loop. From this loop two events might happen: that a new map is received
+  in the queue, or that the timeout from the current map is about to expire and
+  the map needs refreshing. On D-Bus errors, the error message is reported in
+  the corresponding queue and the thread exists.
+
+* `do_main_interactive`: print the selection menu to the user and ask for what
+  action to do next. When the user selects an action, the corresponding map is
+  sent to the que for the D-Bus control thread to handle. It constantly
+  refreshes the terminal to always display the current map on top of the user
+  options menu. If a D-Bus error is received in the corresponding queue, the
+  error message is printed to the user and the thread exists.
+
+## Going to production
+
+This example is meant to educate and show how the Mender D-Bus API works. There
+are a few aspects that you need to take into consideration when writing a
+production grade application based on this example.
+
+### Selecting a UUID
+
+This example uses a dummy [Universally Unique Identifier, or
+UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier). Make sure to
+choose a new random UUID for your application that will stay constant.
+
+### Error handling and failure recovery
+
+The code in this example just prints the message and exits on any D-Bus error. A
+production application shall log the error but try to recover from it. Some of
+the actions that can be taken are retrying at a later time, or reporting the
+error to the user in a better way.
+
+# Refreshing of the maps
+
+The example constantly sets the "current" map. This is necessary for the Mender
+client to keep the map active.
+
+When a map expires it is not taken into account anymore for a Mender update and
+by default the Mender client will continue on any step (assuming there no other
+maps). This is a safety mechanism for the device to always be able to be updated
+when a frontend process has crashed or is hanging.
+
+The application can however customize what happens when a map expires using its
+`on_map_expire` attribute in the map, which can be set to `continue`,
+`force_continue`, or `fail`.
+
+# Update Control Map priority
+
+This example uses priority `-1` in all the maps being set. You may want to use a
+different priority, or a combination of multiple priorities for different maps,
+depending on your application.
+
+When multiple Update Control Maps are active, the Mender client will merge them.
+If multiple maps exist for the same priority, the order of precedence is `fail`,
+`pause`, `force_continue`, else priority takes precedence. `continue` is only
```

Comment:
priority always takes precedence, but for maps with the same priority, it is as you describe here

Reply:
Thanks, I'll rephrase.